### ISSUE TITLE
Stuck with puppeteer v21.0.1

### ISSUE THREAD

**Issue by Amerousful** [GoogleChrome/lighthouse#15588]
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://github.com 

### What happened?

Run just stucks. After a while, I found that the reason is puppeteer version in this [commit](https://github.com/GoogleChrome/lighthouse/pull/15331/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R169).
puppeteer 20.8.0 - works
puppeteer 21.0.1 - doesn't work

### What did you expect?

That it will work...

### What have you tried?

I tested `10.4.0` vs `11.0.0`. In `10.4.0` everythin okay, but in `11.0.0` it stucks forever.
When I used lighthouse programmatically I overrided puppeteer version, and then it works fine.

I used [this](https://github.com/sitespeedio/docker-browsers/blob/main/Dockerfile) Docker image for testings. It already has Chrome, and you need install only particular version
Command for reproducing: 
```bash
lighthouse --chrome-flags="--headless=new --no-sandbox --ignore-certificate-errors --disable-dev-shm-usage" https://github.com
```

<img width="1512" alt="image 749" src="https://github.com/GoogleChrome/lighthouse/assets/22199881/f60e058c-19cb-4acd-94e0-f480c8312ecc">
<img width="1512" alt="image 750" src="https://github.com/GoogleChrome/lighthouse/assets/22199881/d8451405-b248-4f93-b804-2b194a28385b">

### How were you running Lighthouse?

CLI

### Lighthouse Version

11.0.0

### Chrome Version

107.0.5304.62

### Node Version

20.9.0

### OS

Linux 29b727053b9e 6.4.16-linuxkit #1 SMP PREEMPT Wed Oct 25 16:32:24 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux

**Comment by adamraine** [comment]
I can't reproduce outside Docker so it's possibly something in the Docker environment. Does this reproduce with the latest version of Lighthouse (v11.3.0)?

**Comment by Amerousful** [comment]
Yep, seems like it releates with Docker environment. I tried on Mac, and didn't caught this bug.
Yes, it does, it reproduces with 11.3.0, because puppeteer still here. When I used programatically 11.3.0 I just overrided puppeteer to 20.8.0, and it works
